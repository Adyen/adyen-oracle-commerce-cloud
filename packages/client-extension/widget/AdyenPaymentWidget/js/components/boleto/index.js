import * as constants from '../../constants'
import {
    addDays,
    Checkout,
    createPresentToShopperModal,
    eventEmitter,
    createFromAction,
} from '../../utils'
import { store } from '../index'

export const setBoletoConfig = ({
    paymentMethodTypes,
    boletoDeliveryDate,
    boletoShopperStatement,
}) => {
    const hasBoleto = paymentMethodTypes.includes(
        constants.paymentMethodTypes.invoice
    )
    eventEmitter.store.emit(constants.boleto.enabled, hasBoleto)

    if (hasBoleto) {
        boletoDeliveryDate &&
            eventEmitter.store.emit(
                constants.boleto.deliveryDate,
                boletoDeliveryDate
            )
        boletoShopperStatement &&
            eventEmitter.store.emit(
                constants.boleto.shopperStatement,
                boletoShopperStatement
            )
    }
}

export const presentToShopper = customPaymentProperties => {
    const checkoutBoleto = store.get(constants.checkout.boleto)
    const options = {
        action: customPaymentProperties,
        selector: '#present-shopper',
        checkoutComponent: checkoutBoleto,
    }

    createPresentToShopperModal(() => {
        createFromAction(options)
    })
}

const createBoletoCheckout = () => {
    const checkout = new Checkout(constants.paymentMethodTypes.invoice)
    const onChange = checkout.onChange({
        deliveryDate: addDays(store.get(constants.boleto.deliveryDate)),
        shopperStatement: store.get(constants.boleto.shopperStatement),
    })

    const onSubmit = checkout.onSubmit()

    const configuration = { onSubmit, onChange }
    const order = store.get(constants.order)

    const address = order().billingAddress()

    const shopperName = {
        firstName: address.firstName(),
        lastName: address.lastName(),
    }
    const billingAddress = {
        city: address.city(),
        country: address.selectedCountry(),
        // houseNumberOrName:"952",
        postalCode: address.postalCode(),
        stateOrProvince: address.selectedState(),
        street: address.address1(),
    }
    const options = { data: { shopperName, billingAddress } }
    const checkoutOptions = {
        configuration,
        selector: '#adyen-boleto-payment',
        type: 'boletobancario',
        options,
    }

    checkout.createCheckout(checkoutOptions, checkout => {
        eventEmitter.store.emit(constants.checkout.boleto, checkout)
    })
}

export default createBoletoCheckout
