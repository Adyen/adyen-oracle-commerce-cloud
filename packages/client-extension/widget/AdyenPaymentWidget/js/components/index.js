export { default as Component } from './component'
export { default as Order } from './order'
export { default as Payment } from './payment'
export { default as store } from './store'
export {
    default as createBoletoCheckout,
    presentToShopper,
    setBoletoConfig,
} from './boleto'
export { default as createCardCheckout } from './card'
