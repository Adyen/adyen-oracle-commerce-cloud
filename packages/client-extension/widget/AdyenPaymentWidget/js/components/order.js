import $ from 'jquery'
import ccRestClient from 'ccRestClient'
import ccConstants from 'ccConstants'
import storageApi from 'storageApi'
import pubsub from 'pubsub'
import {
    redirectAuth,
    createSpinner,
    getOrderPayload,
    eventEmitter,
} from '../utils'
import * as constants from '../constants'
import { store } from './index'

class Order {
    constructor() {
        this.instance = storageApi.getInstance()

        this.startEventListeners()
    }

    startEventListeners = () => {
        eventEmitter.order.on(
            constants.pageChanged,
            this.getUrlParametersAndCreateOrder
        )
        eventEmitter.order.on(
            constants.initialOrderCreated,
            this.initialOrderCreated
        )
    }

    initialOrderCreated = orderEvent => {
        const order = store.get(constants.order)
        const {
            customPaymentProperties: { data, resultCode },
        } = orderEvent.order.payments[0]

        const payload = {
            order: getOrderPayload(order),
            customPaymentProperties: JSON.parse(data),
            resultCode,
        }

        eventEmitter.store.emit(constants.orderPayload, payload)
        const isDone = store.get(constants.isDone)
        !isDone() && redirectAuth(payload, this.createOrder)
    }

    getAndCreatOrder = result => {
        result.paymentData = this.instance.getItem(
            constants.storage.paymentData
        )

        const payment = { type: 'generic', customProperties: result }

        this.recreateOrder(payment)
    }

    createOrder = orderPayload => {
        const translate = store.get(constants.translate)
        const orderFail = ({ message = 'Failed to create order' } = {}) => {
            $.Topic(pubsub.topicNames.ORDER_SUBMISSION_FAIL).publish({
                message: 'fail',
                errorMessage: message,
            })
        }

        const submitOrder = ({ id, uuid }) => {
            const publishData = { message: 'success', id, uuid }
            $.Topic(pubsub.topicNames.ORDER_COMPLETED).publish(publishData)
            $.Topic(pubsub.topicNames.ORDER_SUBMISSION_SUCCESS).publish([
                publishData,
            ])
        }

        const orderSuccess = data => {
            const { paymentState } = data.payments[0]
            const isAuthorized =
                paymentState === ccConstants.PAYMENT_GROUP_STATE_AUTHORIZED
            const message = translate(
                constants.errorMessages.failed3dsValidation
            )
            isAuthorized ? submitOrder(data) : orderFail({ message })
        }

        ccRestClient.request(
            ccConstants.ENDPOINT_ORDERS_CREATE_ORDER,
            orderPayload,
            orderSuccess,
            orderFail
        )
    }

    recreateOrder = paymentData => {
        const storedOrder = JSON.parse(
            this.instance.getItem(constants.storage.order)
        )
        storedOrder.payments = [paymentData]

        eventEmitter.store.emit(constants.isDone, true)
        this.createOrder(storedOrder)
    }

    getResult = urlParameters => {
        const params = urlParameters.split('&')
        const formatResult = (acc, param) => {
            const [key, value] = param.split('=')
            return { ...acc, [key]: decodeURIComponent(value) }
        }

        return params.reduce(formatResult, {})
    }

    getUrlParametersAndCreateOrder = pageData => {
        const order = store.get(constants.order)
        const { parameters } = pageData

        if (parameters) {
            createSpinner()

            const parametersObj = this.getResult(parameters)

            if (order().paymentGateway()) {
                order().paymentGateway().type = ''
            }

            this.isAction = 'PaRes' in parametersObj && 'MD' in parametersObj
            this.isAction && this.getAndCreatOrder(parametersObj)
        }
    }
}

export default Order
