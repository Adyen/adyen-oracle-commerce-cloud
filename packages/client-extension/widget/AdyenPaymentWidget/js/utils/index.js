export { createSpinner, destroySpinner } from './spinner'
export { default as ajax } from './ajax'
export { default as redirectAuth } from './redirectAuth'
export { default as getOrderPayload } from './getOrderPayload'
export { default as eventEmitter } from './events'
export { default as getInstallmentOptions } from './getInstallmentOptions'
export { default as Checkout, createFromAction } from './checkout'
export { createPresentToShopperModal } from './modal'
export { default as createError } from './createError'
export { addDays } from './date'
