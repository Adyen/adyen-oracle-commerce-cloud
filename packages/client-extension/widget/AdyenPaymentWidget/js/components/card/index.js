import { store } from '../index'
import * as constants from '../../constants'
import { Checkout, eventEmitter, getInstallmentOptions } from '../../utils'

export const onBrand = ({ brand }) => {
    eventEmitter.store.emit(constants.selectedBrand, brand)
    const installmentsOptions = store.get(constants.installmentsOptions)
    const cart = store.get(constants.cart)
    const amount = cart().amount()
    const installments = installmentsOptions ? getInstallmentOptions(installmentsOptions, amount, brand) : []

    eventEmitter.store.emit(constants.installments, installments)
}

export const onConfigSuccess = () => eventEmitter.store.emit(constants.isLoaded, true)

const createCardCheckout = () => {
    // const [card] = paymentMethods.filter(paymentMethod => paymentMethod.type === constants.paymentMethodTypes.scheme)

    // if (card) {
    //     eventEmitter.store.emit(constants.paymentMethodTypes.scheme, true)
    //     const { brands } = card
    const checkout = new Checkout(constants.paymentMethodTypes.scheme)
    const onSubmit = checkout.createOnSubmit()
    const onChange = checkout.createOnChange()
    const cardComponent = document.querySelector('adyen-payment-method-card')
    cardComponent.addEventListener('adyenChange', onChange)
    cardComponent.addEventListener('adyenSubmit', onSubmit)

    eventEmitter.store.emit(constants.checkout.card, checkout)
    // const cardComponent = document.querySelector('adyen-payment-method-card')

    //     const onChange = checkout.onChange()
    //     const onSubmit = checkout.onSubmit()

        // const storedPaymentType = store.get(constants.storedPaymentType)
        // const options = { hasHolderName: false, enableStoreDetails: !!storedPaymentType(), brands }

        // const configuration = { onBrand, onConfigSuccess, onChange, onSubmit }
        // const checkoutOptions = { configuration, selector: '#adyen-card-payment', type: 'card', options }

        // checkout.createCheckout(checkoutOptions, checkout => {
        //     eventEmitter.store.emit(constants.checkout.card, checkout)
        // })
    // }
}

export default createCardCheckout
