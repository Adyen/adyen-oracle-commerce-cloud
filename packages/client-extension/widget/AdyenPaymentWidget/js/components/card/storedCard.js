import { Checkout, eventEmitter } from '../../utils'
import * as constants from '../../constants'

const createStoredCards = () => {
    const checkout = new Checkout(constants.paymentMethodTypes.scheme)
    const onChange = checkout.onChange()
    const configuration = { onChange, onSubmit: checkout.onSubmit(onChange) }
    const checkoutComponent = checkout.getCheckout(configuration)

    eventEmitter.store.emit(
        constants.storedPaymentMethods,
        checkoutComponent.paymentMethodsResponse.storedPaymentMethods
    )

    checkoutComponent.paymentMethodsResponse.storedPaymentMethods.forEach(storedPaymentMethod => {
        const selector = `#adyen-stored_${storedPaymentMethod.id}-payment`
        checkoutComponent.create(constants.card, storedPaymentMethod).mount(selector)
    })
}

export default createStoredCards
