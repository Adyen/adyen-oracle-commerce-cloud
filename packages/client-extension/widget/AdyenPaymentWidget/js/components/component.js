import * as constants from '../constants'
import { eventEmitter } from '../utils'
import { createBoletoCheckout, createCardCheckout, store } from './index'

class Component {
    constructor() {
        this.startEventListeners()
    }

    startEventListeners = () => {
        eventEmitter.component.on(constants.render, this.render)
        eventEmitter.component.on(
            constants.comboCardOptions,
            this.getComboCardOptions
        )
    }

    getComboCardOptions = () => {
        const brands = { visa: constants.bins.electron }
        const selectedComboCard = store.get(constants.selectedComboCard)()
        const isDebitCard = selectedComboCard === constants.comboCards.debit
        const brand = store.get(constants.selectedBrand)
        const selectedBrand =
            brand in brands
                ? brands[store.get(constants.selectedBrand)]
                : store.get(constants.selectedBrand)

        const options = {
            selectedBrand,
            additionalData: { overwriteBrand: true },
        }

        eventEmitter.store.emit(
            constants.comboCardOptions,
            isDebitCard ? options : {}
        )
    }

    render = () => {
        store.get(constants.ajax)('paymentMethods', this.getPaymentMethods)
    }

    getPaymentMethods = paymentMethodsResponse => {
        eventEmitter.store.emit(
            constants.paymentMethodsResponse,
            paymentMethodsResponse
        )

        createCardCheckout()
        store.get(constants.brazilEnabled) && createBoletoCheckout()
    }
}

export default Component
