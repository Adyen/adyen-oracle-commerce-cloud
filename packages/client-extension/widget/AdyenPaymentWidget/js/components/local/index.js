import * as constants from '../../constants'
import { Checkout, eventEmitter } from '../../utils'

const hasType = paymentMethod => 'type' in paymentMethod
const isNotScheme = paymentMethod => hasType(paymentMethod) && paymentMethod.type !== 'scheme'
const hasDetails = paymentMethod => isNotScheme(paymentMethod) && 'details' in paymentMethod
const hasOneItem = paymentMethod => hasDetails(paymentMethod) && paymentMethod.details.length === 1
const getDetail = paymentMethod => hasOneItem(paymentMethod) && paymentMethod.details[0]
const isValid = (paymentMethod, key, value) => getDetail(paymentMethod) && getDetail(paymentMethod)[key] === value

const isLocalPaymentMethod = paymentMethod => {
    const isIssuer = isValid(paymentMethod, 'key', 'issuer')
    const isSelect = isValid(paymentMethod, 'type', 'select')
    const isLocal = isIssuer && isSelect

    return isLocal
}

const createLocalComponent = ({ type, name }) => {
    const headingNode = document.createElement('div')
    headingNode.classList.add('collapse-heading')
    headingNode.innerHTML = ` <strong>
        <span
            class="collapse-toggle collapsed"
            data-toggle="collapse"
            data-target="#adyen-${type}-wrapper"
        >
          <button>${name}</button>
        </span>
      </strong>`
    const component = document.createElement('div')
    component.setAttribute('id', `adyen-${type}-wrapper`)
    component.classList.add('adyen-wrapper', 'collapse')

    component.innerHTML = `
        <div class="loader-wrapper hide">
            <div class="lds-ring"><div></div><div></div><div></div><div></div></div>
            <div style="font-weight: bold; font-size: 16px;">Processing...</div>
        </div>
        <div id="adyen-${type}-payment" class="adyen-component"></div>
    `
    const adyenNode = document.getElementById('adyen-collapse-heading')
    adyenNode.appendChild(headingNode)
    adyenNode.appendChild(component)
}

const createLocalCheckout = (acc, localPaymentMethod) => {
    const { type } = localPaymentMethod
    const checkout = new Checkout(type)
    const onChange = checkout.onChange()
    const onSubmit = checkout.onSubmit()

    const configuration = { onSubmit, onChange, showPayButton: true }

    createLocalComponent(localPaymentMethod)
    const checkoutOptions = type => ({ configuration, selector: `#adyen-${type}-payment`, type })
    checkout.createCheckout(checkoutOptions(type), checkout => ({ ...acc, [type]: checkout }))
}

const createLocalPaymentCheckout = ({ paymentMethods }) => {
    const localPaymentMethods = paymentMethods.filter(isLocalPaymentMethod)
    const localPaymentMethodsCheckout = localPaymentMethods.reduce(createLocalCheckout, {})

    eventEmitter.store.emit(constants.checkout.local, localPaymentMethodsCheckout)
}

export default createLocalPaymentCheckout
