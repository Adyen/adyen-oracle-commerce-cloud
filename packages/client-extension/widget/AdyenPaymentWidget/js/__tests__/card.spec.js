import * as constants from '../constants'
import Widget from '../../../../__mocks__/widget'
import viewModel from '../adyen-checkout'
import paymentMethodsResponseMock from '../../../../__mocks__/paymentMethods'

jest.mock('../utils/checkout')
jest.mock('../utils/getInstallmentOptions')
import { Checkout, eventEmitter, getInstallmentOptions } from '../utils'
import { createCardCheckout, createStoredCards, store } from '../components'
import { onBrand } from '../components/card'
import generateTemplate from '../utils/tests/koTemplate'

describe('Card', () => {
    let widget
    beforeEach(() => {
        widget = new Widget()
    })

    it('should create card checkout', function() {
        viewModel.onLoad(widget)

        Checkout.prototype.createCheckout = jest.fn()
        Checkout.prototype.onChange = jest.fn()
        Checkout.prototype.onSubmit = jest.fn()

        createCardCheckout(paymentMethodsResponseMock)
        expect(Checkout.prototype.createCheckout).toHaveBeenCalled()
    })

    it('should create stored cards checkout', function() {
        viewModel.onLoad(widget)

        Checkout.prototype.createStoredCardCheckout = jest.fn()
        Checkout.prototype.onChange = jest.fn()
        Checkout.prototype.onSubmit = jest.fn()

        createStoredCards()
        expect(Checkout.prototype.createStoredCardCheckout).toHaveBeenCalled()
    })

    it('should set brand', function() {
        const brand = 'visa'
        onBrand({ brand })
        const selectedBrand = store.get(constants.selectedBrand)
        const installments = store.get(constants.installments)

        expect(selectedBrand).toEqual(brand)
        expect(installments()).toEqual([])
    })

    it('should set brand with installments', function() {
        const brand = 'visa'
        eventEmitter.store.emit(constants.installmentsOptions, 'mocked_installments_options')
        onBrand({ brand })

        const expected = ['mocked_installments_options', '1000', 'visa']
        expect(getInstallmentOptions).toHaveBeenCalledWith(...expected)
    })

    it('should display installments', function() {
        const {
            countries: { mexico },
            paymentMethodTypes,
        } = constants
        widget.setLocale(mexico.locale)
        widget.setGatewaySettings('paymentMethodTypes', [paymentMethodTypes.boleto])

        const template = generateTemplate(widget, viewModel => {
            eventEmitter.store.emit(constants.installments, [{ numberOfInstallments: 3 }, { numberOfInstallments: 5 }])
            eventEmitter.store.emit(constants.isLoaded, true)
        })

        expect(template).toMatchSnapshot()
    })

    it('should display combo card', function() {
        const { countries, paymentMethodTypes } = constants
        const { brazil } = countries
        widget.setCurrencyCode(brazil.currency)
        widget.setLocale(brazil.locale)
        widget.setGatewaySettings('paymentMethodTypes', [paymentMethodTypes.scheme])

        const template = generateTemplate(widget, () => {
            eventEmitter.store.emit(constants.installments, [])
            eventEmitter.store.emit(constants.isLoaded, true)
        })
        expect(template).toMatchSnapshot()
    })

    it('should display stored cards', function() {
        const template = generateTemplate(widget, () => {
            const storedPaymentMethods = [{ brand:"visa", expiryMonth:"10", expiryYear:"2020", holderName:"Checkout Shopper PlaceHolder", id:"8415785841327892", lastFour:"1111", name:"VISA", supportedShopperInteractions:["Ecommerce","ContAuth"], type:"scheme"}]
            eventEmitter.store.emit(constants.storedPaymentMethods, storedPaymentMethods)
            eventEmitter.store.emit(constants.isLoaded, true)
        })
        expect(template).toMatchSnapshot()
    })
})
