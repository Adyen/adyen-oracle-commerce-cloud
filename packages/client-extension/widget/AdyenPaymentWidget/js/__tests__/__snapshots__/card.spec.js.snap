// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should display combo card 1`] = `
Object {
  Symbol(dom): <body>
    <div
      class="loader-wrapper hide"
    >
      <div
        class="lds-ring"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
      <div
        style="font-weight: bold; font-size: 16px;"
      >
        Processing...
      </div>
    </div>
    <legend />
    <div
      class="adyen-collapse-heading oc-panel panel-2"
      data-oc-id="panel-2"
    />
    <br />
    <legend />
    <div
      class="adyen-collapse-heading oc-panel panel-1"
      data-oc-id="panel-1"
    >
      <div
        class="collapse-heading"
      >
        <strong>
           
          <span
            class="collapse-toggle collapsed"
            data-target="#adyen-scheme-wrapper"
            data-toggle="collapse"
          >
            <button />
          </span>
        </strong>
      </div>
      <div
        class="adyen-wrapper collapse"
        id="adyen-scheme-wrapper"
      >
        <div
          class="adyen-component"
          id="adyen-card-payment"
        >
          <div
            class="adyen-checkout__field-wrapper"
          >
            <div
              class="adyen-checkout__field"
            >
              <label
                class="adyen-checkout__label"
              >
                <span
                  class="adyen-checkout__label__text"
                />
                <span
                  class="adyen-checkout__input-wrapper"
                >
                  <select
                    aria-required="true"
                    class="adyen-checkout__input adyen-checkout__input--small adyen-checkout__select"
                    id="comboCard"
                    style="-webkit-appearance: none;-moz-appearance: none;appearance: none;background: url('http://cdn1.iconfinder.com/data/icons/cc_mono_icon_set/blacks/16x16/br_down.png') white no-repeat 98.5% !important;background: url('http://cdn1.iconfinder.com/data/icons/cc_mono_icon_set/blacks/16x16/br_down.png') white no-repeat calc(100% - 10px) !important;"
                    type="text"
                  >
                    <option
                      value="creditCard"
                    >
                      translated_creditCard
                    </option>
                    <option
                      value="debitCard"
                    >
                      translated_debitCard
                    </option>
                  </select>
                </span>
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
}
`;

exports[`Card should display installments 1`] = `
Object {
  Symbol(dom): <body>
    <div
      class="loader-wrapper hide"
    >
      <div
        class="lds-ring"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
      <div
        style="font-weight: bold; font-size: 16px;"
      >
        Processing...
      </div>
    </div>
    <legend />
    <div
      class="adyen-collapse-heading oc-panel panel-2"
      data-oc-id="panel-2"
    />
    <br />
    <legend />
    <div
      class="adyen-collapse-heading oc-panel panel-1"
      data-oc-id="panel-1"
    >
      <div
        class="collapse-heading"
      >
        <strong>
           
          <span
            class="collapse-toggle collapsed"
            data-target="#adyen-scheme-wrapper"
            data-toggle="collapse"
          >
            <button />
          </span>
        </strong>
      </div>
      <div
        class="adyen-wrapper collapse"
        id="adyen-scheme-wrapper"
      >
        <div
          class="adyen-component"
          id="adyen-card-payment"
        >
          <div
            class="adyen-checkout__field-wrapper"
          >
            <div
              class="adyen-checkout__field"
            >
              <label
                class="adyen-checkout__label"
              >
                <span
                  class="adyen-checkout__label__text"
                />
                <span
                  class="adyen-checkout__input-wrapper"
                >
                  <select
                    aria-required="true"
                    class="adyen-checkout__input adyen-checkout__input--small adyen-checkout__select"
                    id="installments"
                    style="-webkit-appearance: none;-moz-appearance: none;appearance: none;background: url('http://cdn1.iconfinder.com/data/icons/cc_mono_icon_set/blacks/16x16/br_down.png') white no-repeat 98.5% !important;background: url('http://cdn1.iconfinder.com/data/icons/cc_mono_icon_set/blacks/16x16/br_down.png') white no-repeat calc(100% - 10px) !important;"
                    type="text"
                  >
                    <option
                      value=""
                    >
                      translated_noInstallmentsMsg
                    </option>
                    <option
                      value=""
                    >
                      3x
                    </option>
                    <option
                      value=""
                    >
                      5x
                    </option>
                  </select>
                </span>
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
}
`;

exports[`Card should display stored cards 1`] = `
Object {
  Symbol(dom): <body>
    <div
      class="loader-wrapper hide"
    >
      <div
        class="lds-ring"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
      <div
        style="font-weight: bold; font-size: 16px;"
      >
        Processing...
      </div>
    </div>
    <legend />
    <div
      class="adyen-collapse-heading oc-panel panel-2"
      data-oc-id="panel-2"
    >
      <div
        class="collapse-heading"
      >
        <strong>
          <span
            class="collapse-toggle collapsed"
            data-target="#adyen-stored_8415785841327892-wrapper"
            data-toggle="collapse"
          >
            <button>
              Card ending with 1111
            </button>
          </span>
        </strong>
      </div>
      <div
        class="adyen-wrapper collapse"
        id="adyen-stored_8415785841327892-wrapper"
      >
        <div
          class="adyen-component"
          id="adyen-stored_8415785841327892-payment"
        />
      </div>
    </div>
    <br />
    <legend />
    <div
      class="adyen-collapse-heading oc-panel panel-1"
      data-oc-id="panel-1"
    >
      <div
        class="collapse-heading"
      >
        <strong>
           
          <span
            class="collapse-toggle collapsed"
            data-target="#adyen-scheme-wrapper"
            data-toggle="collapse"
          >
            <button />
          </span>
        </strong>
      </div>
      <div
        class="adyen-wrapper collapse"
        id="adyen-scheme-wrapper"
      >
        <div
          class="adyen-component"
          id="adyen-card-payment"
        />
      </div>
    </div>
  </body>,
}
`;
